### ISSUE TITLE
Sundays as free days?

### ISSUE THREAD

**Issue by navijo** [jsGanttImproved/jsgantt-improved#68]
Hi, 
first of all, congratulations for your plugin. It is awesome in most aspects.
Secondly, I have a question: Is there any way to get sundays as free days and don't count them on the duration?

Thanks!

**Comment by mariohmol** [comment]
Nowdays is counting all days right? this is a good suggestion! thanks

**Comment by mariohmol** [comment]
Can u suggest how to change this function to make it work?
* https://github.com/jsGanttImproved/jsgantt-improved/blob/develop/src/task.ts#L307

Thinking that we have a config like 
* https://github.com/jsGanttImproved/jsgantt-improved/blob/develop/src/draw.ts#L48

**Comment by geeksilva97** [comment]
I do not know if I understood correctly, but I think this is about not having unworked days, like weekends, right?

If so, I've already made this implementation and can post here.

Taking advantage, I apologize to @mariohmol  for the delay in the implementation of **click events**, I have been very busy, but now I will contribute more, with this excellent repository.

**Comment by minasm** [comment]
Hello guys, this is a bit old topic but wondering if this was sorted?
If this not easy to implement maybe you may give us a 'pTaskDuration' field on the object. So we can supply the duration.
Because on some charts we'd like to use as default (count weekends) and some of them we want to keep only 5 days.

Added `pTaskDuration` as an example.
```
g.AddTaskItemObject({
  pID: 1,
  pName: "Define Chart <strong>API</strong>",
  pStart: "2017-08-05",
  pEnd: "2017-08-16",
  pPlanStart: "2017-08-05",
  pPlanEnd: "2017-08-16",
  pTaskDuration:"10 days"
  pClass: "ggroupblack",
  pLink: "",
  pMile: 0,
  pRes: "Brian",
  pComp: 0,
  pGroup: 0,
  pParent: 0,
  pOpen: 1,
  pDepend: "",
  pCaption: "",
  pCost: 1000,
  pNotes: "Some Notes text",
  category: "My Category",
  sector: "Finance"
});
```

**Comment by mariohmol** [comment]
Hi. but there is a pDuration field.. that does that.. no?

**Comment by mariohmol** [comment]
@geeksilva97 do u still have the implementation for it?  thanks!

**Comment by geeksilva97** [comment]
Yes I have, I've been allocated a lot of time on some projects. I will send a PR with an implementation for this feature.

**Comment by mariohmol** [comment]
kkkk thats ok mate =)

**Comment by geeksilva97** [comment]
How can I see the changes I made?

What's port for 

`npm run watch`
?

**Comment by mariohmol** [comment]
run 2 terminals.. one with npm run watch and other with npm run start

**Comment by mariohmol** [comment]
Resolved by #396 (merged). You can now configure which days are working days via `setWorkingDays()` — for example, to exclude Sundays and Saturdays:

```js
g.setWorkingDays({ 0: false, 1: true, 2: true, 3: true, 4: true, 5: true, 6: false });
```

Duration labels and task bar widths/offsets will then reflect only working days. The default is all days enabled, so existing charts are unaffected.
